frontend/src/metabase/embedding-sdk/components/SimpleDataPicker/SimpleDataPicker.tsx

```diff
@@ -1,80 +1,90 @@
-import { useState } from "react";
-import { t } from "ttag";
+import { useDisclosure } from "@mantine/hooks";
+import { type ReactNode, useMemo } from "react";
 
-import { Icon, NavLink, Paper, ScrollArea, TextInput } from "metabase/ui";
-import type { TableId } from "metabase-types/api";
+import { useSearchQuery } from "metabase/api";
+import { isModel } from "metabase/browse/models/utils";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import { Box, Popover } from "metabase/ui";
+import { getQuestionVirtualTableId } from "metabase-lib/v1/metadata/utils/saved-questions";
+import type { SearchResult, TableId } from "metabase-types/api";
+import { SortDirection } from "metabase-types/api/sorting";
+
+import { SimpleDataPickerView } from "./SimpleDataPickerView";
 
 interface SimpleDataPickerProps {
   selectedEntity?: TableId;
-  options: Options[];
-  onClick: (option: any) => void;
-}
-
-interface Options {
-  id: number;
-  display_name: string;
+  isInitiallyOpen: boolean;
+  triggerElement: ReactNode;
+  setSourceTableFn: (tableId: TableId) => void;
 }
 
-const TEN_OPTIONS_HEIGHT = 10 * 33;
-
 export function SimpleDataPicker({
   selectedEntity,
-  options,
-  onClick,
+  isInitiallyOpen,
+  setSourceTableFn,
+  triggerElement,
 }: SimpleDataPickerProps) {
-  const shouldShowSearchBar = options.length > 10;
-  const [searchText, setSearchText] = useState("");
-  function filterSearch(option: Options): boolean {
-    if (searchText) {
-      return normalizeString(option.display_name).includes(
-        normalizeString(searchText),
-      );
+  const [isDataPickerOpened, { toggle, close }] =
+    useDisclosure(isInitiallyOpen);
+  const { data, isLoading, error } = useSearchQuery({
+    models: ["dataset", "table"],
+  });
+
+  const options = useMemo(() => {
+    if (!data) {
+      return [];
     }
 
-    return true;
-  }
+    return sortEntities(
```

Comment:
there's no BE sort for the search endpoint. The sorting is determined by ranking https://github.com/metabase/metabase/blob/b4a34ab899caf2438509b5b15c371f8c41933d32/src/metabase/search/config.clj#L87-L99

And the model browse page sort the result on FE, so I do the same here.